ckan/public-midnight-blue/base/javascript/toast.js

```diff
@@ -0,0 +1,192 @@
+(function (ckan, $) {
+
+  /**
+   * Displays Bootstrap 5 toast notifications across CKAN
+   *
+   * Usage:
+   *
+   * ckan.toast({
+   *    message: "Operation completed successfully",
+   *    type: "success",
+   *    title: "Done",
+   *    icon: "<i class='fa fa-check-circle me-2'></i>",
+   *    subtitle: "Just now",
+   *    delay: 5000,
+   *    position: "bottom-right"
+   *    stacking: false
+   * });
+   *
+   *
+   * Options:
+   * - message   (string)  [required] The main message body of the toast.
+   * - type      (string)  Optional.  Defines toast style, matches keys from `styles`. Default: "default".
+   * - title     (string)  Optional.  Text shown in the toast header.
+   * - icon      (string)  Optional.  HTML for an icon displayed before the title.
+   * - subtitle  (string)  Optional.  Text shown in the header's small subtitle area.
+   * - delay     (number)  Optional.  Time in milliseconds before auto-hide. Default: 3000.
+   * - position  (string)  Optional.  Position key controlling toast placement. Default: "bottom-right".
+   * - stacking  (boolean) Optional.  Whether to stack toasts on top of each other. Default: true.
+  */
+  var toast = function (options) {
+    if (!options || !options.message) {
+      console.error("Toast: Missing required 'message' option.");
+      return;
+    }
+
+    const opts = {
+      type: "default",
+      title: "",
+      icon: "",
+      subtitle: "",
+      delay: 3000,
+      position: "bottom-right",
+      showProgress: true,
```

Comment:
is enabling a progress bar too visually busy as a default?

Please attach a capture of how this looks so we can discuss.

Reply:
Yes, it works, I tested it on my iPad. My code just adds a progress bar and manages its animation. The bootstrap toast has their events to prevent toast from hiding.